File: ClientHandler.java
import java.io.*;
import java.net.*;

class ClientHandler implements Runnable {
    private Socket socket;

    public ClientHandler(Socket socket) {
        this.socket = socket;
    }

    @Override
    public void run() {
        try (BufferedReader input = new BufferedReader(new InputStreamReader(socket.getInputStream()));
             PrintWriter output = new PrintWriter(socket.getOutputStream(), true)) {
            String message;
            output.println("Hello, you are connected to the server!");
            while ((message = input.readLine()) != null) {
                System.out.println("Received: " + message);
                if (message.equalsIgnoreCase("hello")) {
                    output.println("Hello from the server!");
                } else {
                    output.println("Unknown message");
                }
            }
        } catch (IOException e) {
            e.printStackTrace();
        } finally {
            try {
                socket.close();
            } catch (IOException e) {
                e.printStackTrace();
            }
        }
    }
}

File: client-handler.js
class ClientHandler {
    constructor(ws) {
      this.socket = ws;
  
      // Handle incoming messages
      this.socket.on('message', (message) => {
        console.log(`Received message: ${message}`);
        // You can handle messages here
      });
  
      // Handle connection close
      this.socket.on('close', () => {
        console.log('Client disconnected');
      });
    }
  
    // Function to send a message to the client
    sendMessage(message) {
      if (this.socket && this.socket.readyState === this.socket.OPEN) {
        this.socket.send(message);
      }
    }
  }
  
  module.exports = ClientHandler;